packages/flutter/test/material/app_bar_theme_test.dart

```diff
@@ -64,38 +82,77 @@ void main() {
         brightness: brightness,
         elevation: elevation,
         iconTheme: iconThemeData,
-        textTheme: textTheme,)
-      ),
+        actionsIconTheme: actionsIconThemeData,
+        textTheme: textTheme,
+        actions: <Widget>[
+          IconButton(icon: const Icon(Icons.share), onPressed: () {})
+        ],
+      )),
     ));
 
     final Material widget = _getAppBarMaterial(tester);
     final IconTheme iconTheme = _getAppBarIconTheme(tester);
+    final IconTheme actionsIconTheme = _getAppBarActionsIconTheme(tester);
+    final RichText actionIconText = _getAppBarIconRichText(tester);
     final DefaultTextStyle text = _getAppBarText(tester);
 
     expect(SystemChrome.latestStyle.statusBarBrightness, brightness);
     expect(widget.color, color);
     expect(widget.elevation, elevation);
     expect(iconTheme.data, iconThemeData);
+    expect(actionsIconTheme.data, actionsIconThemeData);
+    expect(actionIconText.text.style.color, actionsIconThemeData.color);
     expect(text.style, textTheme.body1);
   });
 
+  testWidgets('AppBar icon color takes priority over everything', (WidgetTester tester) async {
+    const Color color = Colors.lime;
+    const IconThemeData iconThemeData = IconThemeData(color: Colors.green);
+    const IconThemeData actionsIconThemeData = IconThemeData(color: Colors.lightBlue);
+
+    final ThemeData themeData = _themeData().copyWith(appBarTheme: _appBarTheme());
+
+    await tester.pumpWidget(MaterialApp(
+      theme: themeData,
+      home: Scaffold(appBar: AppBar(
+        backgroundColor: color,
```

Comment:
This doesn't seem relevant for the test? Setting backgroundColor to a random color might be a good idea in case somehow the backgroundColor ended being used for the icon's color, but using the expected icon color here seems to be tempting fate.

Reply:
You're right, this was unintended and came from copying the previous test and missing this line when adjusting it. Removed